inst/markdown/compareScenarios2/cs2_main.Rmd

```diff
@@ -70,40 +70,78 @@ if (interactive()) {
 library(gridExtra)
 options(tidyverse.quiet = TRUE)
 library(tidyverse)
+library(kableExtra)
 library(quitte)
 library(mip)
 ```
 
 
-```{r read mifs}
-# Read *.mif-files.
-tibble(path = unname(params$mifScen)) %>% 
-  mutate(
-    newScenarioName = params$mifScenNames,
-    data = map(path, read.quitte, factors=FALSE)) %>% 
-  unnest(cols = data) ->
-  dataScen
-if ('newScenarioName' %in% colnames(dataScen)) {
-  dataScen %>% 
-    mutate(scenario = newScenarioName) ->
-    dataScen
-} 
-dataScen %>% 
+```{r read scenario mifs}
+# Read *.mif-files as nested tibble.
+tibble(path = unname(params$mifScen)) %>%
+  rowid_to_column("fileid") %>%
+  mutate(data = map(path, read.quitte, factors = FALSE)) %>%
+  unnest(data) %>%
+  nest(data = -c(fileid, path, scenario)) ->
+  dataScenNested
+# Add column newScenarioName either with contents of params$mifScenNames or copy names from column scenario.
+if (is.null(params$mifScenNames)) {
+  dataScenNested %>%
+    mutate(newScenarioName = scenario) ->
+    dataScenNested
+} else {
+  dataScenNested %>%
+    left_join(
+      tibble(
+        fileid = seq_along(params$mifScen),
+        newScenarioName = params$mifScenNames),
+      by = "fileid") ->
+    dataScenNested
+}
+# Add suffixes "_1", "_2", ... in case of duplicated scenario names.
+# Side note: In the unlikly case, one has scenario names
+#   scenA, scenA, scenA_1, scenB,
+# they will be renamed to
+#   scenA_1, scenA_2, scenA_1, scenB
+# in the first iteration of the following loop. After the second iteration we get
+#   scenA_1_1, scenA_2, scenA_1_2, scenB.
+while (anyDuplicated(dataScenNested$newScenarioName)) {
+  warning("There are dupicated scenario names. They will be renamed.")
```

Comment:
dupicated -> duplicated :)

Great you thought about this possibility with getting a duplicated name through the replacement process! An alternative solution would be to add `_1_` or `.1` to the scenario names, as scenario names never end with `_` and never contain a `.`, see https://github.com/remindmodel/remind/blob/develop/start.R#L262. But: minor stuff, all good!

Reply:
My code is not really a super clean solution and you made a good point, I haven't thought about these naming restrictions. Unfortunately, I allowed the user to set newScenarioNames without such restrictions. (The restrictions could be easily implemented, though.) So, a user could specify names ending in `_` or containing a `.`. 
But if a user is specifying such adversarial scenario names as in my example by hand, they must be evil anyway...
One could also use `base::make.unique()`, which would probably be the cleanest coding solution, but it transforms `c("a", "a")` to `c("a", "a_1")` instead of `c("a_1", "a_2")` which is a little less nice.
To not waste too much time on a problem that will probably never actually surface, I will just keep it how it is for now, I think.

Reply:
I agree. The make.unique solution would have been ok for me as well. And as I said, great that you thought about this problem in the first place.